### ISSUE TITLE
Server error when creating a new proposal component in BO

### ISSUE THREAD

**Issue by Quentinchampenois** [alecslupu-pfa/decidim-module-custom_proposal_states#12]
#### Description

On the branch `chore/fix-module-dependency`, as an administrator, I can't create a new proposal component.

#### How to reproduce

* Log in as admin
* Create a new Proposal component on a Participatory Process
* Save the form
* See internal server error

#### Stacktrace

```
NoMethodError (undefined method `create_default_states!' for Decidim::Proposals:Module):
  
/Users/User/decidim-module-custom_proposal_states/lib/decidim/custom_proposal_states/engine.rb:128:in `block (3 levels) in <class:Engine>'
decidim (9f17d2920c50) decidim-core/lib/decidim/component_manifest.rb:113:in `block in run_hooks'
decidim (9f17d2920c50) decidim-core/lib/decidim/component_manifest.rb:112:in `map'
decidim (9f17d2920c50) decidim-core/lib/decidim/component_manifest.rb:112:in `run_hooks'
decidim (9f17d2920c50) decidim-admin/app/commands/decidim/admin/create_component.rb:48:in `run_hooks'
decidim (9f17d2920c50) decidim-admin/app/commands/decidim/admin/create_component.rb:25:in `block in call'
activerecord (6.0.6.1) lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `block in transaction'
activerecord (6.0.6.1) lib/active_record/connection_adapters/abstract/transaction.rb:280:in `block in within_new_transaction'
activesupport (6.0.6.1) lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
activesupport (6.0.6.1) lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
activesupport (6.0.6.1) lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
activesupport (6.0.6.1) lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
activesupport (6.0.6.1) lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
activerecord (6.0.6.1) lib/active_record/connection_adapters/abstract/transaction.rb:278:in `within_new_transaction'
activerecord (6.0.6.1) lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `transaction'
activerecord (6.0.6.1) lib/active_record/transactions.rb:212:in `transaction'
rectify (0.13.0) lib/rectify/command.rb:41:in `transaction'
decidim (9f17d2920c50) decidim-admin/app/commands/decidim/admin/create_component.rb:23:in `call'
rectify (0.13.0) lib/rectify/command.rb:30:in `call'
decidim (9f17d2920c50) decidim-admin/app/controllers/decidim/admin/components_controller.rb:36:in `create'
actionpack (6.0.6.1) lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
actionpack (6.0.6.1) lib/abstract_controller/base.rb:195:in `process_action'
actionpack (6.0.6.1) lib/action_controller/metal/rendering.rb:30:in `process_action'
actionpack (6.0.6.1) lib/abstract_controller/callbacks.rb:42:in `block in process_action'
activesupport (6.0.6.1) lib/active_support/callbacks.rb:112:in `block in run_callbacks'
activesupport (6.0.6.1) lib/active_support/core_ext/time/zones.rb:66:in `use_zone'
decidim (9f17d2920c50) decidim-core/app/controllers/concerns/decidim/use_organization_time_zone.rb:21:in `use_organization_time_zone'
activesupport (6.0.6.1) lib/active_support/callbacks.rb:121:in `block in run_callbacks'
i18n (1.14.1) lib/i18n.rb:322:in `with_locale'
decidim (9f17d2920c50) decidim-core/app/controllers/concerns/decidim/locale_switcher.rb:24:in `switch_locale'
activesupport (6.0.6.1) lib/active_support/callbacks.rb:121:in `block in run_callbacks'
activesupport (6.0.6.1) lib/active_support/callbacks.rb:139:in `run_callbacks'
actionpack (6.0.6.1) lib/abstract_controller/callbacks.rb:41:in `process_action'
actionpack (6.0.6.1) lib/action_controller/metal/rescue.rb:22:in `process_action'
actionpack (6.0.6.1) lib/action_controller/metal/instrumentation.rb:33:in `block in process_action'
activesupport (6.0.6.1) lib/active_support/notifications.rb:180:in `block in instrument'
activesupport (6.0.6.1) lib/active_support/notifications/instrumenter.rb:24:in `instrument'
activesupport (6.0.6.1) lib/active_support/notifications.rb:180:in `instrument'
actionpack (6.0.6.1) lib/action_controller/metal/instrumentation.rb:32:in `process_action'
actionpack (6.0.6.1) lib/action_controller/metal/params_wrapper.rb:245:in `process_action'
activerecord (6.0.6.1) lib/active_record/railties/controller_runtime.rb:27:in `process_action'
actionpack (6.0.6.1) lib/abstract_controller/base.rb:136:in `process'
actionview (6.0.6.1) lib/action_view/rendering.rb:39:in `process'
actionpack (6.0.6.1) lib/action_controller/metal.rb:190:in `dispatch'
actionpack (6.0.6.1) lib/action_controller/metal.rb:254:in `dispatch'
actionpack (6.0.6.1) lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
actionpack (6.0.6.1) lib/action_dispatch/routing/route_set.rb:33:in `serve'
actionpack (6.0.6.1) lib/action_dispatch/journey/router.rb:49:in `block in serve'
actionpack (6.0.6.1) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.0.6.1) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.0.6.1) lib/action_dispatch/routing/route_set.rb:834:in `call'
railties (6.0.6.1) lib/rails/engine.rb:527:in `call'
```

#### Extra

I think method `create_default_states!` is called on the wrong object here : https://github.com/alecslupu-pfa/decidim-module-custom_proposal_states/blob/7f915dc83a5bebcfaaed06d90e9ba05e9a67de40/lib/decidim/custom_proposal_states/engine.rb#L128

If it is the case I can open a PR if needed, thanks !

**Comment by alecslupu** [comment]
@Quentinchampenois  this is fixed in 5a01dcd0c61f7760adb929d53a6ccdbe6af5b050

Please confirm.

**Comment by Quentinchampenois** [comment]
I confirm this is fixed

Closing this issue, thank you !
